test/Prototypes/PatternMatching.swift

```diff
@@ -35,18 +35,16 @@ protocol Pattern {
   associatedtype MatchData = ()
   
   func matched<C: Collection>(atStartOf c: C) -> MatchResult<Index, MatchData>
-  where C.Index == Index, Element_<C> == Element
+  where C.Index == Index, C.Element == Element
```

Comment:
There are other usages of `Element_` remaining, are they still needed?

Reply:
Good point, @slavapestov!  I'll address those too.

Reply:
Done.